go/logger/context_with_logging.go

```diff
@@ -0,0 +1,58 @@
+package logger
+
+import (
+	"testing"
+
+	"golang.org/x/net/context"
+)
+
+type ContextAndLogger struct {
+	ctx context.Context
+	Logger
+}
+
+var _ CtxAndLogger = ContextAndLogger{}
```

Comment:
The go convention is that the interface ends in `-er` and the instantiation doesn't.  It doesn't always work out. I'm still not a fan of `CtxAndLogger` versus `ContextAndLogger`. That's very unclear.